compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1113,12 +1114,27 @@ public void visit(BLangTypeInit cIExpr) {
                 break;
             case TypeTags.STREAM:
             case TypeTags.CHANNEL:
-                if (cIExpr.initInvocation.argExprs.size() > 0) {
+                if (!cIExpr.initInvocation.argExprs.isEmpty()) {
                     dlog.error(cIExpr.pos, DiagnosticCode.TOO_MANY_ARGS_FUNC_CALL, cIExpr.initInvocation.name);
                     resultType = symTable.semanticError;
                     return;
                 }
                 break;
+            case TypeTags.UNION:
+                List<BType> matchingMembers = findMembersWithMatchingInitFunc(cIExpr, (BUnionType) actualType);
+                BType matchedType = getMatchingType(matchingMembers, cIExpr, actualType);
+
+                expType = matchedType;
+                if (matchedType.tag == TypeTags.OBJECT
```

Comment:
Checking TypeTag.Object is enough, because, we know that, the matched type should have an initializer function from previous validations. isn't it ?

Reply:
Ahh, there can be a scenario, the user hasn't mentioned the init function. In that case we need to check.